util/drakeUtil.cpp

```diff
@@ -108,17 +109,20 @@ mxArray* createDrakeMexPointer(void* ptr, const char* name, int num_additional_i
 
 void* getDrakeMexPointer(const mxArray* mx)
 {
-	void* ptr = NULL;
+  void* ptr = NULL;
 
-	// todo: optimize this by caching the pointer values, as described in
-	// http://groups.csail.mit.edu/locomotion/bugs/show_bug.cgi?id=1590
-	mxArray* ptrArray = mxGetProperty(mx,0,"ptr");
-	if (!ptrArray)
-		mexErrMsgIdAndTxt("Drake:getDrakeMexPointer:BadInputs","cannot retrieve 'ptr' field from this mxArray.  are you sure it's a valid DrakeMexPointer object?");
+  if (!mxIsClass(mx, "DrakeMexPointer")) 
+    mexErrMsgIdAndTxt("Drake:getDrakeMexPointer:BadInputs", "getDrakeMexPointer can only be called on arguments which correspond to DrakeMexPointer objects");
 
-  if (!mxIsNumeric(ptrArray) || mxGetNumberOfElements(ptrArray)!=1)
+  // todo: optimize this by caching the pointer values, as described in
+  // http://groups.csail.mit.edu/locomotion/bugs/show_bug.cgi?id=1590
+  mxArray* ptrArray = mxGetProperty(mx, 0, "ptr");
+  if (!ptrArray)
+    mexErrMsgIdAndTxt("Drake:getDrakeMexPointer:BadInputs", "cannot retrieve 'ptr' field from this mxArray.  are you sure it's a valid DrakeMexPointer object?");
+
+  if (!mxIsNumeric(ptrArray) || mxGetNumberOfElements(ptrArray) != 1)
```

Comment:
i think the most important check is that it's the proper size uint, instead of a double? (though i don't want to call mexCallMatlab for isa)